src/app.ts

```diff
@@ -41,7 +41,7 @@ async function main(): Promise<void> {
     return;
   }
 
-  browser = await launch({ userDataDir: ROOT_PATH + '/data', headless: true });
+  browser = await launch({ userDataDir: ROOT_PATH + '/data', headless: true, defaultViewport: null, args: ['--window-size=1,1'] });
```

Comment:
@shihabmridha

Reply:
Will take a look. Thanks.

Reply:
**Strange!** I did not encounter any issue. Removed the **data** directory to remove session etc. Then ran the script with `loginCheck: true`. Got logged in and downloaded the course without any issue.

Reply:
I am not sure but for me it's not getting logged in. Let me share the error it throws

Reply:
`⠹ Loggin in.Node is either not visible or not an HTMLElement
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! educative.io-downloader@1.0.0 start: `node build/src/app.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the educative.io-downloader@1.0.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/kshitij/.npm/_logs/2020-04-21T14_18_50_409Z-debug.log`

Reply:
this is happening only after adding my changes, without it,  it's working fine. But my changes are required in order to capture full code.

Reply:
A quick solution would be launch a browser without your change (defaultViewport, args) to login. Then close the browser and open a new one with your changes.

I see an old issue related to this (https://github.com/puppeteer/puppeteer/issues/2977) as well.

BTW, Which OS are you using? What is the screen resolution of your display?

Reply:
Using mac, yes that would work but I don't want this manual step. Any other way it can be solved?

Reply:
How about including this step in script only? I mean when login it will be without my changes and then it will close and launch another browser instance with my changes?

Reply:
anywhere we can connect and chat?

Reply:
My Hangouts: shihabmridha@gmail.com

Reply:
Discord username: DIJKSTRA#8375
Sent you an invitation to Hangouts.

Reply:
When tried it is working but downloading almost blank images or pdf or HTML. Do u know what might be the issue?

Reply:
Make sure you have HTML selected. Secondly make headless = false and then try. Which course are you trying to download?

Reply:
@thomasgeoff the main problem I think is. You are not getting logged in. In config file change loginCheck to true so it will check first if you are logged in or not and then will proceed further.

Reply:
I did that and it logged in and started downloading. I tried all the formats. Images only show a broken centre line, pdf's are blank and HTML have only the first page. I'm trying to download grokking the coding interview patterns. Is it working for you?

Reply:
Yes working perfectly for me

Reply:
Im getting this error:
Processing batch of lessons.Evaluation failed: TypeError: Cannot read property 'childNodes' of undefined at __puppeteer_evaluation_script__:4:80

should I install typescript globally?

Reply:
@shihabmridha @devKshitijJain When I try with loginCheck: true. It gives me this error:
> node build/src/app.js

/ Loggin in.Node is either not visible or not an HTMLElement
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! educative.io-downloader@1.0.0 start: `node build/src/app.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the educative.io-downloader@1.0.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\tgeof\AppData\Roaming\npm-cache\_logs\2020-04-24T03_59_13_005Z-debug.log

Reply:
@thomasgeoff this is known issue. Just make headless=false and then try.

Reply:
@thomasgeoff already discussed this with @shihabmridha. Will work on fix.